### PR TITLE
Fix DateTime string generation.

### LINKED ISSUE #479: Transform object to yaml and back has problem
Pom as following:

            <dependency>
              <groupId>io.kubernetes</groupId>
              <artifactId>client-java</artifactId>
              <version>3.0.0</version>
            </dependency>

I am using io.kubernetes.client.util.Yaml class to do the transformation from object and back. Codes as following.

    V1Pod pod = coreV1Api.readNamespacedPod(podName, namespace, null, null, null);
    String yaml = Yaml.dump(pod);
    V1Pod pod2 = Yaml.loadAs(yaml, V1Pod.class)

I encounter the error with line 3 like following:

    org.yaml.snakeyaml.nodes.MappingNode cannot be cast to org.yaml.snakeyaml.nodes.ScalarNode
     in 'reader', line 4, column 22:
          creationTimestamp: {}
                             ^
    
      at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(Constructor.java:270)
      at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.construct(Constructor.java:149)
      at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:207)
      at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:196)
      at io.kubernetes.client.util.Yaml$CustomConstructor.constructObject(Yaml.java:334)
      at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(Constructor.java:247)
      ... 12 more
    Caused by: java.lang.ClassCastException: org.yaml.snakeyaml.nodes.MappingNode cannot be cast to org.yaml.snakeyaml.nodes.ScalarNode
      at io.kubernetes.client.util.Yaml$CustomConstructor.constructObject(Yaml.java:331)
      at org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(Constructor.java:247)
      ... 17 more
    
Here is the snippet of the yaml transformed from object

    apiVersion: v1
    kind: Pod
    metadata:
      creationTimestamp: {}
      generateName: test-776d6c86cc-
      labels:
        app: test
        app-version: '19911'
        pod-template-hash: '3328274277'
      name: test-776d6c86cc-4zwj5
      namespace: default
    ...

Here is the snippet of the yaml when I use `kubectl edit pod xxxx` command:

    apiVersion: v1
    kind: Pod
    metadata:
      creationTimestamp: "2018-12-23T01:09:18Z"
      generateName: test-776d6c86cc-
      labels:
        app: test
        app-version: "19911"
        pod-template-hash: "3328274277"
      name: test-776d6c86cc-4zwj5
      namespace: default

It seems Yaml class has some problem to convert DateTime to String

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #479

### CODE REVIEW

**Comment by brendandburns** [comment]
@cliveseldon any chance I could get a review on this one? I'd like to merge before I cut the next release...

File: util/src/main/java/io/kubernetes/client/util/Yaml.java [hunk 1]
```diff
@@ -358,6 +358,15 @@ public CustomRepresenter() {
       this.representers.put(IntOrString.class, new RepresentIntOrString());
       this.representers.put(byte[].class, new RepresentByteArray());
       this.representers.put(Quantity.class, new RepresentQuantity());
+      this.representers.put(DateTime.class, new RepresentDateTime());
+    }
+
+    private class RepresentDateTime implements Represent {
+      @Override
+      public Node representData(Object data) {
+        DateTime dateTime = (DateTime) data;
```
Comment (cliveseldon) [inline util/src/main/java/io/kubernetes/client/util/Yaml.java]: Why is dateTime cast from data when then not used in next line for conversion to String?

Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/util/Yaml.java]: Removed.

**Comment by brendandburns** [comment]
@cliveseldon Thanks for the review! Comment addressed.

**Pull request merged**